src/components/Signup/AddAccount.tsx

```diff
@@ -1,76 +1,143 @@
 import { useNavigate } from 'react-router-dom';
 import styled, { css } from 'styled-components';
 
-import { SignUpBanner } from '../../asset/image';
+import {
+  EMAIL_ERROR_MSG,
+  EMAIL_REGEX,
+  PASSWORD_CONFIRM_ERROR_MSG,
+  PASSWORD_ERROR_MSG,
+  PASSWORD_REGEX,
+} from '../../constant/signup';
 import { SignUpInfo } from '../../types/signup';
 
 const AddAccount = () => {
   const navigate = useNavigate();
+  const [signupInfo, setSignupInfo] = useState<SignUpInfo>({
+    emailInfo: {
+      email: null,
+      isValid: false,
+      errorMsg: null,
+    },
+    passwordInfo: {
+      password: null,
+      isValid: false,
+      errorMsg: null,
+    },
+    passwordConfirmInfo: {
+      isValid: false,
+      errorMsg: null,
+    },
+  });
+
+  const handleValidation = (inputValueType: string, inputValue: string) => {
+    switch (inputValueType) {
+      case 'email':
+        !EMAIL_REGEX.test(inputValue)
+          ? setSignupInfo({
+              ...signupInfo,
+              emailInfo: { email: null, isValid: false, errorMsg: EMAIL_ERROR_MSG },
+            })
+          : setSignupInfo({
+              ...signupInfo,
+              emailInfo: { email: inputValue, isValid: true, errorMsg: null },
+            });
+        break;
+
+      case 'password':
+        !PASSWORD_REGEX.test(inputValue)
+          ? setSignupInfo({
+              ...signupInfo,
+              passwordInfo: { password: null, isValid: false, errorMsg: PASSWORD_ERROR_MSG },
+            })
+          : setSignupInfo({
+              ...signupInfo,
+              passwordInfo: { password: inputValue, isValid: true, errorMsg: null },
+            });
+        break;
+
+      case 'passwordConfirm':
+        inputValue !== signupInfo.passwordInfo.password
+          ? setSignupInfo({
+              ...signupInfo,
+              passwordConfirmInfo: { isValid: false, errorMsg: PASSWORD_CONFIRM_ERROR_MSG },
+            })
+          : setSignupInfo({
+              ...signupInfo,
+              passwordConfirmInfo: { isValid: true, errorMsg: null },
+            });
+        break;
+    }
+  };
 
-  const {
-    register,
-    handleSubmit,
-    formState: { errors, isValid },
-    getValues,
-  } = useForm<SignUpInfo>({ mode: 'onChange' });
+  const handleSubmitAccount = (e: React.FormEvent) => {
+    e.preventDefault();
 
-  const handleSubmitAccount = () => {
-    const { email, password } = getValues();
-    const signupDataInfo = { email, password };
+    const finalEmail = signupInfo.emailInfo.email;
+    const finalPassword = signupInfo.passwordInfo.password;
+    const signupDataInfo = { email: finalEmail, password: finalPassword };
     navigate(`/signup/nickname`, { state: { signupDataInfo } });
   };
 
+  const handleSpace = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const currentInputValue = e.target.value;
+    if (currentInputValue.includes(' ')) {
+      e.target.value = currentInputValue.replace(' ', '');
+    }
+  };
+
```

Comment:
중간에 공백이 들어갔을 때 커서가 맨 뒤로 가는 이슈 -> input 요소가 제공하는 dom api `selectionStart`, `setSelectionRange`을 사용해서 해결할 수 있어용!!! 
( input type이 email인 경우에는 `selectionStart`가 적용이 안된다구 하는데 지금은 text라서 요걸로 해결 가능할듯 !! )
https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange
```suggestion
  const handleSpace = (e: React.ChangeEvent<HTMLInputElement>) => {
    const currentInput = e.target;
    const currentInputValue = e.target.value;
    if (currentInputValue.includes(' ')) {
      const position = currentInput.selectionStart && currentInput.selectionStart - 1;
      e.target.value = currentInputValue.replace(' ', '');
      currentInput.setSelectionRange(position, position);
    }
  };

```

Reply:
미띤 대박~!~! 아 커서 위치 조정 치니까 바로 나오네,,, Input 공백 커서 위치,,,이런거나 찾고 잇었어 
역시 검색도 실력이다 고마어 바로 반영함~!~!